src/main/java/com/denizenscript/denizencore/scripts/commands/core/MongoCommand.java

```diff
@@ -0,0 +1,442 @@
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+                    DenizenCore.schedule(new AsyncSchedulable(new OneTimeSchedulable(() -> {
+                        String conStr = uri.getValue();
+                        if (!conStr.startsWith("mongodb://") && !conStr.startsWith("mongodb+srv://")) {
+                            conStr = "mongodb://" + conStr;
+                        }
+                        Debug.echoDebug(scriptEntry, "Connecting to Mongo server...");
+                        MongoClient con = null;
+                        MongoDatabase db = null;
+                        MongoCollection<Document> col = null;
+                        try {
+                            con = MongoClients.create(conStr);
+                            db = con.getDatabase(database);
+                            col = db.getCollection(collection);
+                        }
+                        catch (final Exception e) {
+                            DenizenCore.runOnMainThread(() -> {
+                                Debug.echoError(scriptEntry, "Mongo Exception: " + e.getMessage());
+                                scriptEntry.setFinished(true);
+                                if (CoreConfiguration.debugVerbose) {
+                                    Debug.echoError(scriptEntry, e);
+                                }
+                            });
+                        }
+                        if (CoreConfiguration.debugVerbose) {
+                            Debug.echoDebug(scriptEntry, "Connection did not error.");
+                        }
+                        final MongoClient conn = con;
+                        final MongoDatabase connDB = db;
+                        final MongoCollection<Document> coll = col;
+                        if (con != null) {
+                            DenizenCore.runOnMainThread(() -> {
+                                Connection mongoConnection = new Connection();
+                                mongoConnection.connection = conn;
+                                mongoConnection.database = connDB;
+                                mongoConnection.collection = coll;
+                                mongoConnections.put(connectionId, mongoConnection);
+                                Debug.echoDebug(scriptEntry, "Successfully connected to Mongo server.");
+                                scriptEntry.setFinished(true);
+                            });
+                        }
+                        else {
+                            DenizenCore.runOnMainThread(() -> {
+                                Debug.echoDebug(scriptEntry, "Connecting errored!");
+                                scriptEntry.setFinished(true);
+                            });
+                        }
+                    }, 0)));
+                    return;
+                }
+                if (connection.connection == null) {

```

Comment:
this will NPE as it's `connection` that will be null, not `connection.connection` - remember to test error cases, not just the good cases